lib/fastifySession.js

```diff
@@ -192,19 +191,14 @@ function session (fastify, options, next) {
 
   function isConnectionSecure (request) {
     return (
-      request.raw.socket?.encrypted === true ||
-    request.headers['x-forwarded-proto'] === 'https'
+      request.raw.socket?.encrypted === true || request.headers['x-forwarded-proto'] === 'https'
     )
   }
 
-  function shouldSaveSession (request, cookieOpts, saveUninitialized) {
-    if (!saveUninitialized && !request.session.isModified()) {
-      return false
-    }
-    if (cookieOpts.secure !== true || cookieOpts.secure === 'auto') {
-      return true
-    }
-    return isConnectionSecure(request)
+  function getShouldSaveSession (request, cookieId, saveUninitializedSession, rollingSessions) {
+    return cookieId !== request.session.encryptedSessionId
+      ? saveUninitializedSession || request.session.isModified()
+      : rollingSessions || (Boolean(request.session.cookie.expires) && request.session.isModified())
```

Comment:
I borrowed this code from `express-session`:
https://github.com/expressjs/session/blob/master/index.js#L464-L473

Interestingly, they have both `shouldSave` and `shouldSetCookie`. This distinction will be important later when we correctly implement `session.touch()`